spec/integration_helper.rb

```diff
@@ -0,0 +1,38 @@
+require 'active_model'
+
+class Price
+  extend EncapsulateAsMoney
+
+  case ActiveModel::VERSION::MAJOR
+  when 3 then include ActiveModel::AttributeMethods
+  when 4 then include ActiveModel::Model
+  else raise NotImplementedError, ActiveModel::VERSION::MAJOR
+  end
+
+  include ActiveModel::Validations
+
+  encapsulate_as_money :amount
+  validates_inclusion_of :amount, in: Money.new(1)..Money.new(2), allow_nil: true
+  validates_inclusion_of :qty, in: 1..2, allow_nil: true
+
+  def initialize(attributes = {})
+    @amount = attributes[:amount]
+    @qty = attributes[:qty]
+  end
+
+  def amount
+    @amount
+  end
+
+  def amount=(amount)
+    @amount = amount
+  end
```

Comment:
The gem calls `super`. I could have simulated the more common use case with `ActiveRecord` but this was enough to expose the bug which needed fixing.